File: virtual-world-simulator/src/Entity/Plants/Dandelion.cpp
#include "Dandelion.h"

Dandelion::Dandelion(World* world, Point position) :
	Plant { world, position, 'D', 0, SpreadChance::LOW }
{
}

void Dandelion::update()
{
	for(int i = 0; i < 3; i++)
	{
		int random = Utility::random(1, 25);
		if (random <= this->spreadChance)
			this->spread<Dandelion>();
	}

	Entity::update();
}

bool Dandelion::collision(Entity& entity)
{
	this->isAlive = false;
    return true;
}

File: Java/virtual-world-simulator/src/Entities/Plants/Dandelion.java
package Entities.Plants;

import App.World;
import Common.Point;

import javax.swing.*;

public class Dandelion extends Plant
{
    public Dandelion(World world, Point position)
    {
        super(new ImageIcon("assets/dandelion.png"), world, position, 'D', 0, 0, SpreadChance.LOW);
    }

    public void Update()
    {
        super.Update();

        for(int i = 0; i < 3; i++)
            Spread(new Grass(this.world, new Point(-1, -1)));
    }
}
